splink/postgres/postgres_linker.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import annotations
+
+import logging
+
+import pandas as pd
+from sqlalchemy import text
+from sqlalchemy.engine import Engine
+
+from ..input_column import InputColumn
+from ..linker import Linker
+from ..misc import ensure_is_list
+from ..splink_dataframe import SplinkDataFrame
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresDataFrame(SplinkDataFrame):
+    linker: PostgresLinker
+
+    def __init__(self, df_name, physical_name, linker):
+        super().__init__(df_name, physical_name, linker)
+        self._db_schema = "splink"
+        self.physical_name = f"{self.physical_name}"
+
+    @property
+    def columns(self) -> list[InputColumn]:
+        sql = f"""
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE table_name = '{self.physical_name}';
+        """
+        res = self.linker._run_sql_execution(sql).fetchall()
+        cols = [r["column_name"] for r in res]
+
+        return [InputColumn(c, sql_dialect="postgres") for c in cols]
+
+    def validate(self):
+        if type(self.physical_name) is not str:
+            raise ValueError(
+                f"{self.df_name} is not a string dataframe.\n"
+                "Postgres Linker requires input data"
+                " to be a string containing the name of the"
+                " postgres table."
+            )
+
+        sql = f"""
+        SELECT table_name
+        FROM information_schema.tables
+        WHERE table_name = '{self.physical_name}';
+        """
+
+        res = self.linker._run_sql_execution(sql).fetchall()
+        if len(res) == 0:
+            raise ValueError(
+                f"{self.physical_name} does not exist in the postgres db provided.\n"
+                "Postgres Linker requires input data"
+                " to be a string containing the name of a"
+                " postgres table that exists in the provided db."
+            )
+
+    def drop_table_from_database(self, force_non_splink_table=False):
+        self._check_drop_table_created_by_splink(force_non_splink_table)
+        self.linker._delete_table_from_database(self.physical_name)
+
+    def as_record_dict(self, limit=None):
+        sql = f"""
+        SELECT *
+        FROM {self.physical_name}
+        """
+        if limit:
+            sql += f" LIMIT {limit}"
+        sql += ";"
+        res = self.linker._run_sql_execution(sql).mappings().all()
+        return res
+
+
+class PostgresLinker(Linker):
+    def __init__(
+        self,
+        input_table_or_tables,
+        settings_dict=None,
+        engine: Engine = None,
+        set_up_basic_logging=True,
+        input_table_aliases: str | list = None,
+    ):
+        self._sql_dialect_ = "postgres"
+        # TODO: allow this to happen under hood?
```

Comment:
Yeah I think that is useful - had a couple of thoughts but couldn't quite settle on what the best interface is, and felt like maybe it could serve as an upgrade. Will raise an issue.